aeron-driver/src/test/c/aeron_strutil_test.cpp

```diff
@@ -42,7 +46,7 @@ TEST_F(StrUtilTest, shouldHandleSingleValue)
     EXPECT_EQ(num_tokens, 1);
     EXPECT_STREQ(tokens[0], "single_token");
 
-    aeron_free(input);
+    free(input);
```

Comment:
Why not keep the `aeron_free` version and have it do the right thing depending on platform?

Reply:
`aeron_free` already does the right thing and calls `free`. I'm not sure why, but this additional indirection causes an assertion in the debug build. It looks like the two different modules use two different heaps.

In any case, I think `free` is more correct here, as the `strdup` docs tell you should call `free` on what it returns. Something like `aeron_free` could potentially switch to a different allocator in the future, breaking this code even for other platforms.

Reply:
`aeron_free` should be used by all memory calls. The debug build is probably mixing new and malloc.

Reply:
The code uses `strdup` so we can't really control how it allocates the memory. The docs simply say to use `free` to release it.
Under a MSVC debug build, going through `aeron_free` causes an assertion failure, while using `free` works fine.

Reply:
As `aeron_free` uses free under the covers.... it indicates a different problem is at play. So, just moving to `free` does not fix that. Also, as I mentioned, all memory free should use `aeron_free`. Probably shouldn't be using `strdup` either. @mikeb01 can these not use `strdup` and be on the stack instead so that we can get rid of both `free` and `aeron_free` in this test?

Reply:
I've removed the strdup and free from the test as of 22401b35b

Reply:
> can these not use `strdup` and be on the stack instead so that we can get rid of both `free` and `aeron_free` in this test?

I like that solution, it should be easy with a `strcpy` to the stack.

Reply:
@tmontgomery OK I'm pretty sure I understand the problem now.

CMake generated different configurations regarding the CRT library for the media driver and the unit test: the media driver uses *"Multi-threaded Debug DLL (/MDd)"* while the test uses *"Multi-threaded Debug (/MTd)"*.

This basically means they use a different instance of the C runtime, so memory allocated by one cannot be freed by the other.

Anyway, your fix makes the issue go away.

Reply:
AH. Makes perfect sense. Glad you figured it out and mentioned it. And glad the change works. Thanks!